tests/abi/functions.h

```diff
@@ -0,0 +1,273 @@
+                            uint16_t _03,
+                            uint16_t _04,
+                            uint16_t _05,
+                            uint16_t _06,
+                            uint16_t _07,
+                            uint16_t _08,
+                            uint16_t _09,
+                            uint16_t _10,
+                            uint16_t _11,
+                            uint16_t _12,
+                            uint16_t _13,
+                            uint16_t _14,
+                            uint16_t _15,
+                            uint16_t _16) {
+  PRINT_HEADER(functionWithLotsOfArguments);
+  puts("  Arguments:");
+  PRINT_UINT16_T(_00);
+  PRINT_UINT16_T(_01);
+  PRINT_UINT16_T(_02);
+  PRINT_UINT16_T(_03);
+  PRINT_UINT16_T(_04);
+  PRINT_UINT16_T(_05);
+  PRINT_UINT16_T(_06);
+  PRINT_UINT16_T(_07);
+  PRINT_UINT16_T(_08);
+  PRINT_UINT16_T(_09);
+  PRINT_UINT16_T(_10);
+  PRINT_UINT16_T(_11);
+  PRINT_UINT16_T(_12);
+  PRINT_UINT16_T(_13);
+  PRINT_UINT16_T(_14);
+  PRINT_UINT16_T(_15);
+  PRINT_UINT16_T(_16);
+  puts("  ReturnValues:");
+  puts("");
+}
+
+void __attribute__((noinline, weak)) ABIDEF
+multitypeArgumentFunction(int64_t _00,
+                          uint32_t _01,
+                          int16_t _02,
+                          uint8_t _03,
+                          __int128_t _04) {
+  PRINT_HEADER(multitypeArgumentFunction);
+  puts("  Arguments:");
+  PRINT_INT64_T(_00);
+  PRINT_UINT32_T(_01);
+  PRINT_INT16_T(_02);
+  PRINT_UINT8_T(_03);
+  PRINT_INT128_T(_04);
+  puts("  ReturnValues:");
+  puts("");
+}
+
+void __attribute__((noinline, weak)) ABIDEF
+functionForRandomnessTesting(uint8_t _00,
+                             uint8_t _01,
+                             uint8_t _02,
+                             uint8_t _03,
+                             uint8_t _04,
+                             uint8_t _05,
+                             uint8_t _06,
+                             uint8_t _07,
+                             uint8_t _08,
+                             uint8_t _09,
+                             uint8_t _10,
+                             uint8_t _11,
+                             uint8_t _12,
+                             uint8_t _13,
+                             uint8_t _14,
+                             uint8_t _15,
+                             uint8_t _16) {
+  PRINT_HEADER(functionForRandomnessTesting);
+  puts("  Arguments:");
+  PRINT_UINT8_T(_00);
+  PRINT_UINT8_T(_01);
+  PRINT_UINT8_T(_02);
+  PRINT_UINT8_T(_03);
+  PRINT_UINT8_T(_04);
+  PRINT_UINT8_T(_05);
+  PRINT_UINT8_T(_06);
+  PRINT_UINT8_T(_07);
+  PRINT_UINT8_T(_08);
+  PRINT_UINT8_T(_09);
+  PRINT_UINT8_T(_10);
+  PRINT_UINT8_T(_11);
+  PRINT_UINT8_T(_12);
+  PRINT_UINT8_T(_13);
+  PRINT_UINT8_T(_14);
+  PRINT_UINT8_T(_15);
+  PRINT_UINT8_T(_16);
+  puts("  ReturnValues:");
+  puts("");
+}
+
+void __attribute__((noinline, weak)) ABIDEF single16bitArgument(uint16_t _00) {
```

Comment:
Maybe create a macro for attribute sets you often use.

Reply:
Done.